File: DataFormManager.Models/FormDataModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataFormManager.Models
{
    public class FormDataModel
    {
        public int? FormId
        {
            get;
            set;
        }
        public string FormType
        {
            get;
            set;
        }
        public string FormName
        {
            get;
            set;
        }
        public List<DataValueModel> FormData
        {
            get;
            set;
        }
        public DatesModel EffectiveDate
        {
            get;
            set;
        }

    }
}

File: DataFormManagerApp/src/app/form-module/models/form-data-model.ts
import { DataValueModel } from './data-value-model';
import { Datesmodel } from './datesmodel';

export class FormDataModel {
    FormId: number;
    FormType: string;
    FormName: string;
    FormData: Array<DataValueModel>;
    EffectiveDate: Datesmodel;
    constructor(formType: string, formName: string, formData: Array<DataValueModel>, effectiveDate: Datesmodel, formId: number = null) {
        this.FormId = formId;
        this.FormType = formType;
        this.FormName = formName;
        this.FormData = formData;
        this.EffectiveDate = effectiveDate;
    }
}
